rocAL/rocAL/include/meta_data/meta_data.h

```diff
@@ -529,10 +544,14 @@ class KeyPointBatch : public BoundingBoxBatch
     }
     std::shared_ptr<MetaDataBatch> clone(bool copy_contents) override
     {
-        if(copy_contents)
+        if(copy_contents) {
             return std::make_shared<KeyPointBatch>(*this);
-        else
-            return std::make_shared<KeyPointBatch>();
+        } else {
+            std::shared_ptr<MetaDataBatch> joints_batch_instance = std::make_shared<KeyPointBatch>();
+            joints_batch_instance->resize(this->size());
+            joints_batch_instance->get_info_batch() = this->get_info_batch();
+            return joints_batch_instance;
+        }
     }
     JointsDataBatch & get_joints_data_batch() { return _joints_data; }
     void copy_data(std::vector<void*> buffer) override {}
```

Comment:
I think for keypoint, we did not update the api for copy_data and get_buffer_size
Please check and update if applicable

Reply:
For Keypoints I have not done any updates yet.
Also does Keypoints storing make use of the ring buffer? @sampath1117

Reply:
Will keypoints data be updated in any of the metanode?

Reply:
Yes keypoint meta data will be updated in flip and warp affine nodes.
For warp affine it is acutally getting updated in node_warp_affine.cpp, since we need to give the updated affine matrix to RPP kernel

Reply:
does Keypoints storing make use of the ring buffer? @sampath1117